src/main/java/org/traccar/protocol/MegastekProtocolDecoder.java

```diff
@@ -276,7 +279,7 @@ private Position decodeOld(Channel channel, SocketAddress remoteAddress, String
             .groupEnd("?").text(",")
             .number("(d+)?,")                    // rfid
             .number("([01])(d)?").optional()     // charge and belt status
-            .expression("[^,]*,").optional()
+            .expression("[^,]*,")
```

Comment:
I've changed this back, as the comma is necessary and the `*` allows no text to be present as well. However, the comma is necessary still.